lib/sidekiq_unique_jobs/client/middleware.rb

```diff
@@ -63,7 +63,7 @@ def old_unique_for?
 
       def new_unique_for?
         connection do |conn|
-          return conn.set(payload_hash, item['jid'], nx: true, ex: expires_at)
+          return conn.set(payload_hash, item['jid'], nx: true, ex: expires_at) || conn.get(payload_hash) == item['jid']
```

Comment:
@deltaroe This is how Sidekiq-unique-jobs always handled jobs before -- it's not a "fix" if you implement a strategy that works for you and breaks it for everyone else.